File: Administrativo/spring-erp-web/SpringNg/src/app/erp_module/shared/provincia/servicio/ProvinciaServicio.ts
import { Component, Inject, Injectable } from '@angular/core';
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Header } from 'primeng/primeng';
import { Provincia } from '../dominio/Provincia';

@Injectable()
export class ProvinciaServicio {

    private rutaServicio: string;
    private http: HttpClient;
    private baseUrl: string;

    constructor(http: HttpClient, @Inject('BASE_URL') baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
        this.rutaServicio = this.baseUrl + 'api/spring/core/Provincia/';
    }

    public listarActivosPorDepartamento(idDepartamento: string): Promise<Provincia[]> {

        const params = new HttpParams()
            .set('idDepartamento', idDepartamento);

        return this.http.get(this.rutaServicio + 'listarActivosPorDepartamento', { params })
            .toPromise()
            .then(response => {
                return response as Provincia[];
            })
            .catch(error => new Array());
    }

    public obtenerNombrePorId(departamento: string, provincia: string): Promise<string> {

        const params = new HttpParams()
            .set('departamento', departamento)
            .set('provincia', provincia);

        return this.http.get(this.rutaServicio + 'obtenerNombrePorId', { params })
            .toPromise()
            .then(response => {
                return (response as any).nombre;
            })
            .catch(error => '');
    }


}

File: Administrativo/spring-erp-servicio/net/royal/spring/core/servicio/ProvinciaServicio.cs
using System;
using System.Collections.Generic;
using System.Text;

using net.royal.spring.framework.web.servicio;
using net.royal.spring.core.dominio;
using net.royal.spring.core.dominio.filtro;

namespace net.royal.spring.core.servicio
{

public interface ProvinciaServicio : GenericoServicio {
        List<Provincia> listarTodos();
        List<Provincia> listar(FiltroProvincia filtro);
        List<Provincia> listarActivosPorDepartamento(String idDepartamento);
        String obtenerNombrePorId(String departamento,String provincia);
    }
}
